tools/bcl-test-importer/BCLTestImporter/BCLTestProjectGenerator.cs

```diff
@@ -417,6 +483,51 @@ bool IsIgnored(BCLTestProjectDefinition project, Platform platform)
 			return projectPaths;
 		}
 		
+		async Task<List<(string name, string path, bool xunit)>> GenerateMacTestProjectsAsync (
+			IEnumerable<(string name, string[] assemblies)> projects, string generatedDir, Platform platform)
+		{
+			var projectPaths = new List<(string name, string path, bool xunit)> ();
+			foreach (var def in projects) {
+				var projectDefinition = new BCLTestProjectDefinition (def.name, def.assemblies);
+				if (IsIgnored (projectDefinition, platform)) // some projects are ignored, so we just continue
+					continue;
+
+				if (!projectDefinition.Validate ())
+					throw new InvalidOperationException ("xUnit and NUnit assemblies cannot be mixed in a test project.");
+				// generate the required type registration info
+				var generatedCodeDir = Path.Combine (generatedDir, projectDefinition.Name);
+				if (!Directory.Exists (generatedCodeDir)) {
```

Comment:
`Directory.CreateDirectory` checks if the directory exists, so this check is redundant.